elastic/kibana
# [ECO][Inventory v2] Ad hoc data view: Add get entities definition endpoint using sources

Closes #202298 
⚠️ Depends on https://github.com/elastic/kibana/pull/203246

## Summary

This PR changes the way we get the entity index patterns to v2. It creates an endpoint part of the inventory API which returns the index patterns by entity type.

## Testing

~⚠️ Currently in order to test we need to create the definition using the dev tools manually before this [PR](https://github.com/elastic/kibana/pull/203246) is merged~ ✅ Merged
 - Open Dev tools and add a definition for an existing entity type (example with host) 
 ```
POST kbn:/internal/entities/v2/definitions/sources
{
  "source": {
    "id": "host_id",
    "type_id": "host",
    "index_patterns": ["metrics-*", "metricbeat-*"],
    "identity_fields": ["host.name"],
    "metadata_fields": [],
    "filters": [],
    "timestamp_field": "@timestamp"
  }
 ```

### Test the endpoint: 
- Open Dev tools and add
` GET kbn:/internal/inventory/entity/definitions/sources?type=host `
- Response: 
<img width="408" alt="image" src="https://github.com/user-attachments/assets/1c7416b0-50a4-4e63-8081-928ec7856ff8">

### Test in the UI
- After the previous steps add some host data (oblt cluster / metricbeat) or use synthtrace (for example use `node scripts/synthtrace infra_hosts_with_apm_hosts --scenarioOpts.numInstances=10`)
- Go to Inventory and expand the host group
- Click on the actions button for any host and click on the Discover link
- The correct dataview should be selected based on the created definition
The same can be done for other entity types

https://github.com/user-attachments/assets/c9c3a7ae-daff-4d4b-b1b7-898c3f1603c6
